src/app/api/community.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable } from '@angular/core';
+import { BaseApiService, Version } from './base-api.service';
+
+export interface Community {
+  name: string;
+  title: string;
+  description: string;
+  sidebar: string;
+  nsfw: boolean;
+  creatorId: number;
+}
+
+export interface CommunityList {
+  communities: Array<string>;
+  totalCount: number;
+  offset: number;
+  size: number;
+  sort: number;
+}
+
+export interface TemplateList {
+  templates: Array<string>;
+  totalCount: number;
+  offset: number;
+  size: number;
+  sort: number;
+}
+
+@Injectable()
+export class CommunityService {
+  // private community: Community;
+  // private name: string;
+
+  constructor(private api: BaseApiService) { }
+
+  // create new community
+  createCommunity(
+    name: string,
```

Comment:
It is probably a better idea to pass in an object with the attributes rather than each parameter here. It'll allow you to make some parameters optional more easily.